src/OpenFOAM/CMakeLists.txt

```diff
@@ -609,4 +609,14 @@ target_link_libraries(OpenFOAM PUBLIC Pstream)
 target_include_directories(OpenFOAM PUBLIC
   .
 )
+
+find_package(ZLIB REQUIRED)
+if(ZLIB_FOUND)
+  add_definitions(-DHAVE_LIBZ)
+endif()
```

Comment:
And this https://github.com/openfoamdotcom/openfoam/blob/d29f00600d42321444580c3d49d167045e01bce4/src/OpenFOAM/CMakeLists.txt#L598 should cover these lines.